src/Sarif.UnitTests/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -162,38 +163,53 @@ private void ValidateResults(string output)
 
-            FileData fileData = FileData.Create(new Uri(uriString), mimeType: "text/x-csharp", encoding: Encoding.UTF8);
-            fileData.Length = 12345;
-
-            FileData outputFileData = visitor.VisitDictionaryValueNullChecked(uriString, fileData);
-            outputFileData.MimeType.Should().Be(fileData.MimeType);
-            outputFileData.Encoding.Should().Be(fileData.Encoding);
-            outputFileData.Length.Should().Be(fileData.Length);
-        }
-
-        [Fact]
-        public void InsertOptionalDataVisitorTests_VisitDictionaryValueNullChecked_InvalidEncoding()
-        {
-            var visitor = new InsertOptionalDataVisitor(OptionallyEmittedData.OverwriteExistingData | OptionallyEmittedData.TextFiles);
-            visitor.VisitRun(new Run()); // VisitDictionaryValueNullChecked requires a non-null run
+            Run run = new Run()
+            {
+                DefaultFileEncoding = "UTF-8",
+                OriginalUriBaseIds = null,
+                Results = new[]
+                {
+                    new Result()
+                    {
+                        Locations = new []
+                        {
+                            new Location
+                            {
+                                PhysicalLocation = new PhysicalLocation
+                                {
+                                     FileLocation = new FileLocation
+                                     {
+                                        Uri = new Uri(inputFileName, UriKind.Relative),
+                                        UriBaseId = uriBaseId
+                                     }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+                        
+            var visitor = new InsertOptionalDataVisitor(OptionallyEmittedData.TextFiles);
+            visitor.VisitRun(run);
 
-            string uriString = "file:///C:/src/foo.cs";
+            run.OriginalUriBaseIds.Should().BeNull();
+            run.Files.Keys.Count.Should().Be(1);
+            run.Files[fileKey].Contents.Should().BeNull();
 
-            FileData fileData = FileData.Create(new Uri(uriString), mimeType: "text/x-csharp");
-            fileData.Encoding = "invalid";
-            fileData.Length = 54321;
+            visitor = new InsertOptionalDataVisitor(OptionallyEmittedData.TextFiles, originalUriBaseIds);
+            visitor.VisitRun(run);
 
-            FileData outputFileData = visitor.VisitDictionaryValueNullChecked(uriString, fileData);
-            outputFileData.MimeType.Should().Be(fileData.MimeType);
-            outputFileData.Encoding.Should().BeNull();
-            outputFileData.Length.Should().Be(fileData.Length);
+            run.OriginalUriBaseIds.Should().Equal(originalUriBaseIds);
+            run.Files[fileKey].Contents.Text.Should().Be(File.ReadAllText(Path.Combine(testDirectory, inputFileName)));
```

Comment:
>File [](start = 57, length = 4)

In a separate PR I advocate injecting an `IFileSystem` into the `InsertOptionalDataVisitor`'s ctor so this can all be mocked and no file system access is needed.